internal/auth/sourcegraphoperator/associate.go

```diff
@@ -0,0 +1,81 @@
+package sourcegraphoperator
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/auth/providers"
+	"github.com/sourcegraph/sourcegraph/internal/auth"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+type accountDetailsBody struct {
+	ClientID  string `json:"clientID"`
+	AccountID string `json:"accountID"`
+
+	ExternalAccountData
+}
+
+// AddSourcegraphOperatorExternalAccount links the given user with a Sourcegraph Operator
+// provider, if and only if it already exists. The provider can only be added through
+// Enterprise Sourcegraph Cloud config, so this essentially no-ops outside of Cloud.
+//
+// 🚨 SECURITY: Some important things to note:
+//   - The caller must check that the user is a site administrator.
+//   - Being a SOAP user does not grant significant privileges over being a site admin.
```

Comment:
```suggestion
//   - Being a SOAP user does not grant any extra privilege over being a site admin.
```

I think this is more accurate.